### PR TITLE
feat: physical optimizer RemoveDuplicate to remove duplicate exec plans

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

A tiny physical rule to remove useless adjoining plans.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

### CODE REVIEW

**Review by evenyag — APPROVED** [review]

File: src/query/src/optimizer/remove_duplicate.rs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        let result = plan
+            .transform_down_mut(&mut |plan| {
+                if plan.as_any().is::<CoalesceBatchesExec>()
+                    || plan.as_any().is::<RepartitionExec>()
+                {
+                    // check child
+                    let child = plan.children()[0].clone();
+                    if child.as_any().type_id() == plan.as_any().type_id() {
+                        // remove child
+                        let grand_child = child.children()[0].clone();
```
Comment (killme2008) [inline src/query/src/optimizer/remove_duplicate.rs:59]: If I don't misunderstand, it looks like every plan node only has one child. Am I right?

**Review by killme2008 — APPROVED** [review]
LGTM.  But I have a few questions, may you take a look? Thank you.

File: src/query/src/optimizer/remove_duplicate.rs (hunk 1 above)
Comment (waynexia) [reply src/query/src/optimizer/remove_duplicate.rs:59]: not *every* plan, but plans referenced here only have one child.

File: src/query/src/optimizer/remove_duplicate.rs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+impl RemoveDuplicate {
+    fn do_optimize(plan: DfPhysicalPlanRef) -> DfResult<DfPhysicalPlanRef> {
+        let result = plan
+            .transform_down_mut(&mut |plan| {
+                if plan.as_any().is::<CoalesceBatchesExec>()
+                    || plan.as_any().is::<RepartitionExec>()
+                {
+                    // check child
+                    let child = plan.children()[0].clone();
```
[the remark this answers is not in the document]
Comment (waynexia) [reply src/query/src/optimizer/remove_duplicate.rs:56]: Yes. Those plan have and only have one child

**Pull request merged**